agent/reconciler/host_reconciler.go

```diff
@@ -171,6 +207,42 @@ func (r *HostReconciler) getBootstrapScript(ctx context.Context, dataSecretName,
 	return bootstrapSecret, nil
 }
 
+func (r *HostReconciler) getInstallationScript(ctx context.Context, dataSecretName, namespace string) (install, uninstall string, err error) {
+	secret := &corev1.Secret{}
+	err = r.Client.Get(ctx, types.NamespacedName{Name: dataSecretName, Namespace: namespace}, secret)
+	if err != nil {
+		return "", "", err
+	}
+
+	installScript := string(secret.Data["install"])
+	uninstallScript := string(secret.Data["uninstall"])
+	return installScript, uninstallScript, nil
+}
+
+func (r *HostReconciler) executeScript(ctx context.Context, script string) error {
+	cmd := exec.CommandContext(ctx, "/bin/bash", "-c", script)
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *HostReconciler) parseInstallationScript(ctx context.Context, script string) (string, error) {
```

Comment:
As it could be used for both
```suggestion
func (r *HostReconciler) parseScript(ctx context.Context, script string) (string, error) {
```